### PR TITLE
addressing #802

### LINKED ISSUE #802: Add optimizations enabled to test_kem and test_sig
The test harnesses test\_kem and test\_sig should output which version of an algorithm's implementation is running.  This may require adding an "optimizations" field to the OQS\_KEM and OQS\_SIG objects.

### PR DESCRIPTION
**Pull request by baentsch**
Fixes #802 

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the the list of algorithms available -- either adding or removing?  (If so, PRs in OQS-OpenSSL, OQS-BoringSSL, and OQS-OpenSSH will also be required by the time this is merged.)

### CODE REVIEW

**Comment by baentsch** [comment]
@dstebila Please check whether this output contains what you had in mind with issue #802:
```
Configuration info
==================
Target platform:  x86_64-Linux-5.4.0-45-generic
Compiler:         gcc (7.5.0)
Compile options:  [-Werror;-Wall;-Wextra;-Wpedantic;-Wstrict-prototypes;-Wshadow;-Wformat=2;-Wfloat-equal;-Wwrite-strings;-O3;-fomit-frame-pointer;-fdata-sections;-ffunction-sections;-Wl,--gc-sections;-Wbad-function-cast]
OQS version:      0.4.0
Git commit:       aa3ef4be20d8eebd414b3a380165422938f3b401 (+ local modifications)
OpenSSL enabled:  Yes (OpenSSL 1.1.1  11 Sep 2018)
AES:              OpenSSL
SHA-2:            OpenSSL
SHA-3:            OpenSSL
CPU exts active:  AES-AVX-n/a-n/a-n/a-n/a-POPCNT-SSE-SSE2-SSE3


```

CI test failures are due to some problems in picnic and doxygen.

File: tests/system_info.c [hunk 1]
```diff
@@ -54,6 +57,35 @@ static void print_platform_info(void) {
...
+
+	OQS_CPU_EXTENSIONS_UNION ext_u;
+	ext_u.ext_x = OQS_get_available_CPU_extensions();
+	printf("CPU exts active:  ");
+	unsigned int it = sizeof(ext_u.ext_a) / sizeof(ext_u.ext_a[0]);
+	for (unsigned int i = 0; i < it; i++) {
+		if (ext_u.ext_a[i]) {
+			printf("%s", OQS_get_cpu_extension_name(i));
+		} else {
+			printf("n/a");
```
Comment (dstebila) [inline tests/system_info.c:77]: I don't know if it's necessary to print "n/a" for ones that are disabled.

**Review by dstebila — APPROVED** [review]

**Comment by dstebila** [comment]
Looks great, thanks Michael!

File: tests/system_info.c (hunk 1 above)
Comment (baentsch) [reply tests/system_info.c]: Agree -- looked weird. Gone.

**Comment by baentsch** [comment]
> Looks great, thanks Michael!

But CI still doesn't quite pass... OK to merge regardless?

**Review by dstebila — APPROVED** [review]

**Pull request merged**